scripts/vet.sh

```diff
@@ -70,6 +70,9 @@ not git grep "\(import \|^\s*\)\"google.golang.org/grpc/interop/grpc_testing" --
 # - Ensure that no trailing spaces are found.
 not git grep '[[:blank:]]$'
 
+# - Ensure that all files have a terminating newline.
+git grep -l '$' -- "*.txt" "*.md" | xargs -I {} sh -c '[ -n "$(tail -c 1 "{}")" ] && echo "{}: No terminating new line found"'
```

Comment:
you have to negate the output. git grep returns exit status 1 when success so we have to negate to get 0. do `not git grep....`

Reply:
We should apply this restriction to all files in our repo I think, not just txt/md files. (We don't check in any binary files, at least not currently, so no need to worry about that.) So the first part can be `git ls-files` instead of a `grep`.

Reply:
Also to prevent your `sh` command from erroring, you can do `&& echo "{}: ..." || true` (but maybe there's a better way?).

Reply:
> We should apply this restriction to all files in our repo I think, not just txt/md files. (We don't check in any binary files, at least not currently, so no need to worry about that.) So the first part can be `git ls-files` instead of a `grep`.

I noticed that the g3 presubmits only mandate terminating newline in text files, so I thought of following that, but I will change it to check in all the files

Reply:
> Also to prevent your `sh` command from erroring, you can do `&& echo "{}: ..." || true` (but maybe there's a better way?).

Added specific checks for`sh` command errors by avoiding running on empty files and checking permissions and readability